prelude/slang-cpp-types.h

```diff
@@ -20,7 +20,8 @@ struct FixedArray
 {
     const T& operator[](size_t index) const { SLANG_PRELUDE_ASSERT(index < SIZE); return m_data[index]; }
     T& operator[](size_t index) { SLANG_PRELUDE_ASSERT(index < SIZE); return m_data[index]; }
-    
+    T* elementPtr(size_t index) { SLANG_PRELUDE_ASSERT(index < SIZE); return m_data + index; }
+
```

Comment:
Should we lower things to use `&array[index]` instead of `array.elementPtr(index)`? Avoiding an explicit `elementPtr` would mean that it is easier to swap out `FixedArray<T,N>` for an actual `T[N]` if we want to (or if we eventually want to target plain C).

Reply:
Done.